File: dotnet/Mammoth/Couscous/mammoth/internal/styles/parsing/LineParseException.cs
using Mammoth.Couscous.java.lang;

namespace Mammoth.Couscous.org.zwobble.mammoth.@internal.styles.parsing {
    internal class LineParseException : RuntimeException {
        private readonly Token _token;
        
        internal LineParseException(Token token, string message) : base(message) {
            _token = token;
        }
        
        internal Token getToken() {
            return _token;
        }
    }
}

File: src/main/java/org/zwobble/mammoth/internal/styles/parsing/LineParseException.java
package org.zwobble.mammoth.internal.styles.parsing;

class LineParseException extends RuntimeException {
    static <T> LineParseException lineParseException(Token<T> token, String message) {
        return new LineParseException(token.getCharacterIndex(), message);
    }

    private final int characterIndex;

    private LineParseException(int characterIndex, String message) {
        super(message);
        this.characterIndex = characterIndex;
    }

    int getCharacterIndex() {
        return characterIndex;
    }
}
